src/auth/providers.js

```diff
@@ -50,13 +44,12 @@ function userPassProvider(auth) {
   return {
     /**
      * Login to a stitch application using username and password authentication
-     *
-     * @param {String} username the username to use for authentication
-     * @param {String} password the password to use for authentication
-     * @returns {Promise}
      */
-    login: (username, password, opts) => {
+    authenticate: data => {
+      const {username, password} = data;
```

Comment:
i think you could just avoid the need for this line by changing L48 to just:
`authenticate : ({username, password}) => ...`